tests/e2e/tests/pilot/performance/serviceentry_test.go

```diff
@@ -0,0 +1,346 @@
+package performance
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/google/uuid"
+	"github.com/onsi/gomega"
+	mcp "istio.io/api/mcp/v1alpha1"
+	networking "istio.io/api/networking/v1alpha3"
+	mixerEnv "istio.io/istio/mixer/test/client/env"
+	"istio.io/istio/pilot/pkg/bootstrap"
+	"istio.io/istio/pilot/pkg/model"
+	mcpserver "istio.io/istio/pkg/mcp/server"
+	"istio.io/istio/pkg/test/env"
+	mockmcp "istio.io/istio/tests/e2e/tests/pilot/mock/mcp"
+	"istio.io/istio/tests/util"
+)
+
+var (
+	serviceEntryCount     = flag.Int("count", 1000, "Number of Service Entries to register")
+	configRefreshInterval = flag.Int("interval", 30, "Number seconds to regenerate a new set of Service Entries")
+	terminateAfter        = flag.Int("after", 240, "Number of seconds to run the test")
+)
+
+const (
+	pilotDebugPort = 5555
+	pilotGrpcPort  = 15010
+	serviceDomain  = "example.com"
+)
+
+func TestServiceEntry(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	test := newTestParam(*serviceEntryCount, *configRefreshInterval, *terminateAfter)
+	go test.init()
+
+	t.Log("building & starting mock mcp server...")
+	mcpServer, err := runMcpServer(test)
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	defer mcpServer.Close()
+
+	tearDown := initLocalPilotTestEnv(t, mcpServer.Port, pilotGrpcPort, pilotDebugPort)
```

Comment:
> try at least 50 concurrent clients for any meaningful result.

I assume you mean creating more of the  mockMCP server?

Reply:
Yes, you can use adsc.go for a more realistic and simpler interface ( there is also a stand alone load tester using it that I'm using quite a bit).

Reply:
I think this comment is referring to line 57 where we create the envoy client(s)? 

What is the goal of this test? Measuring e2e cpu/memory of Pilot would be useful. At the same time, it might be nice to isolate config time (MCP from Galley) from runtime (xDS push to Pilot). That might be possible with parameterization, e.g. reduce # of envoy clients to zero to isolate config time benchmarking.

Reply:
One goal (for me) is to measure the MCP ServiceEntry propagation delay inside pilot.
It doesn't need to include Galley or CF or envoy - we can do that in the end-to-end tests.

Reply:
> This test is to collect cpu and memory profile/usage while registering N number of service entries over t period of time.

The original purpose of this test was to measure CPU/Memory usage. The basic idea was that we could have small focused and targeted benchmarks tests which could be easily integrated in post-submit tests so we can track memory usage over time for the config time path in Pilot. 

Measuring propagation delay is also valuable. Should we include that in this test, or create a separate test for that purpose?

Reply:
The initial goal I had in mind is to measure cpu/memory under a give load (in this case focus is service entry) while keeping this test independent of an specific platform (e.g. Galley, CF) but as for envoy I was under assumption it would be critical to keep envoy for more accurate and realistic samples produced. Having said that I do like the idea of config time isolation (regardless of platform CF, Galley) from runtime by introducing a flag for the # envoys. Also measuring MCP ServiceEntry propagation delay inside pilot is very interesting too, but not too sure how to go ahead about that without using any sort of metrics.

Reply:
> Measuring propagation delay is also valuable. Should we include that in this test, or create a separate test for that purpose?

I would say a separate test might seem cleaner and make the results more useful. Keeping all these tests isolated in terms of responsibly might make them more useful, expandable and easier to follow in the future.